network/p2p/throttler_handler.go

```diff
@@ -20,25 +20,37 @@ var (
 	_            Handler = (*ThrottlerHandler)(nil)
 )
 
+func NewThrottlerHandler(handler Handler, throttler Throttler, log logging.Logger) *ThrottlerHandler {
+	return &ThrottlerHandler{
+		handler:   handler,
+		throttler: throttler,
+		log:       log,
+	}
+}
+
 type ThrottlerHandler struct {
-	Handler
-	Throttler Throttler
-	Log       logging.Logger
+	handler   Handler
+	throttler Throttler
+	log       logging.Logger
 }
 
 func (t ThrottlerHandler) AppGossip(ctx context.Context, nodeID ids.NodeID, gossipBytes []byte) {
-	if !t.Throttler.Handle(nodeID) {
-		t.Log.Debug("dropping message", zap.Stringer("nodeID", nodeID))
+	if !t.throttler.Handle(nodeID) {
+		t.log.Debug("dropping message", zap.Stringer("nodeID", nodeID))
 		return
 	}
 
-	t.Handler.AppGossip(ctx, nodeID, gossipBytes)
+	t.handler.AppGossip(ctx, nodeID, gossipBytes)
 }
 
 func (t ThrottlerHandler) AppRequest(ctx context.Context, nodeID ids.NodeID, deadline time.Time, requestBytes []byte) ([]byte, error) {
-	if !t.Throttler.Handle(nodeID) {
+	if !t.throttler.Handle(nodeID) {
 		return nil, fmt.Errorf("dropping message from %s: %w", nodeID, ErrThrottled)
 	}
 
-	return t.Handler.AppRequest(ctx, nodeID, deadline, requestBytes)
+	return t.handler.AppRequest(ctx, nodeID, deadline, requestBytes)
+}
+
+func (t ThrottlerHandler) CrossChainAppRequest(ctx context.Context, chainID ids.ID, deadline time.Time, requestBytes []byte) ([]byte, error) {
+	return t.handler.CrossChainAppRequest(ctx, chainID, deadline, requestBytes)
```

Comment:
I guess we don't want to handle this case?

Reply:
Yeah... this is the same behavior as before but my thought was that it didn't make sense to throttle messages from ourselves. I think we should probably still throttle anyways, but probably in a future PR. Currently nothing is using this.